packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -2170,6 +2198,17 @@ Hardhat Network's forking functionality only works with blocks from at least spu
         (blockContext.header as any).baseFeePerGas = new BN(0);
       }
 
+      originalCommon = (this._vm as any)._common;
+      (this._vm as any)._common = new Common({
+        chain: {
+          // eslint-disable-next-line @typescript-eslint/dot-notation
+          ...this._vm._common["_chainParams"],
+          chainId: this._configChainId,
+          networkId: this._configNetworkId,
```

Comment:
I think these are incorrect. If the block is a historic one we should be using the remote network's values.

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1928/commits/b1c59f1d82c0c0157d8e90a9416e07c1323f1e3c